dvc/remote/hdfs.py

```diff
@@ -143,8 +143,6 @@ def _download(self, from_info, to_file, **_kwargs):
 
     @contextmanager
     def open(self, path_info, mode="r", encoding=None):
-        assert mode in {"r", "rt", "rb"}
-
```

Comment:
Why did you remove this `assert`? It is still working only for those modes. As I see it this PR should only touch local and ssh.

Reply:
@Suor , the assertion was moved to `repo.open`; I'll revert the commit